pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -191,6 +183,35 @@ func (gc *GarbageCollector) Sync(discoveryClient discovery.ServerResourcesInterf
 		gc.workerLock.Lock()
 		defer gc.workerLock.Unlock()
 
+		newResources, err := gc.resync(discoveryClient, period, stopCh)
```

Comment:
we just did a lot of work to discover newResources, shouldn't we pass that in as the starting point for our resync?

also, assign the result of resync to a new var like `resyncedResources`, recognizing that might be different than what we passed in

Reply:
I agree. I think it would be more readable if you put the for loop outside of `resync`